### PR TITLE
Patient Details report table

### PR DESCRIPTION
**Pull request by peterMuriuki**
Refactor the patient details table to use slice data from the logface slices.

### CODE REVIEW

**Review by moshthepitt — DISMISSED** [review]

**Review by moshthepitt — APPROVED** [review]

File: src/components/PatientDetailsReports/index.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { Dictionary, sortByEventDate } from '../../helpers/utils';
+import { LogFaceSmsType, NutritionLogFaceSms, PregnancyLogFaceSms } from '../../store/ducks/sms_events';
+import React from 'react';
+
+interface PatientDetailsReportProps {
+    patientsReports: (PregnancyLogFaceSms | NutritionLogFaceSms)[];
+    isChild: boolean;
+}
+
+const defaultProps = {
```
Comment (machariamuguku) [inline src/components/PatientDetailsReports/index.tsx:16]: Not a fan of default props

File: src/components/PatientDetailsReports/index.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    patientsReports: [],
+    isChild: false,
+};
+
+export type ReportTableTypes = PatientDetailsReportProps;
```
Comment (machariamuguku) [inline src/components/PatientDetailsReports/index.tsx:21]: why?

**Review by moshthepitt — APPROVED** [review]

File: src/components/PatientDetailsReports/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+                        <p>
+                            <Trans t={t}>Showing reports for:&emsp;</Trans>
+                        </p>
+                        <div className="filters">
+                            <Select
+                                placeholder={t('Select pregnancy')}
+                                options={filterOptions}
+                                value={filterValue}
+                                onChange={(val) => {
+                                    setCurrentPregnancyEventId(val?.value ?? '');
```
Comment (machariamuguku) [inline src/components/PatientDetailsReports/index.tsx:83]: you could check for val.value then set state only when it changes?

**Review by machariamuguku — APPROVED** [review]

File: src/components/PatientDetailsReports/index.tsx (hunk 2 above)
Comment (peterMuriuki) [reply src/components/PatientDetailsReports/index.tsx:21]: will remove this

**Pull request merged**